### PR TITLE
Fix up modelling issues that we’re encountering

### PR DESCRIPTION
**Pull request by ozzyaaron**
I’ve introduced the ISBLANK and COUNTBLANK functions. This allows us to then exclude values depending on whether we have all the required inputs or not. 

eg. IF( COUNTBLANK( sample_inputs ) = 0, sample_outputs, 0 ) can be a common use case. 

Also what I found was that it’s relatively easy to end up up with a sheet that seems to work in Excel but doesn’t work in Soroban. The main reason so far has been that Excel will show an empty string as a 0 if the cell has been set to output numbers. Further calculations based on this number then work. As Soroban uses the base value ( empty string ) then calculations like N129-N128 would fail if N129 or N128 evaluate to a blank string. 

The _simple_ way to fix this is to attempt to coerce cell values to a float where appropriate. This is a blunt fix but will probably work quite well. The more appropriate fix is to handle this in the grammar. The assumption being that an additive or multiplicative operation should only involve numbers and the expression should not be valid with anything but numbers. This fix would require more time than I have right now given we have a practical fix in place with this PR.

### CODE REVIEW

File: lib/soroban/functions/is_blank.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Excel ISBLANK returns true for empty string and null
```
Comment (stwr667) [inline lib/soroban/functions/is_blank.rb:1]: what about for whitespace? e.g. `" "`? That's always the gotcha!

Comment (stwr667) [inline lib/soroban/functions/is_blank.rb:1]: oh wait, I can see that it uses rails `.blank?` so yes of course :D

**Comment by stwr667** [comment]
:+1:

File: lib/soroban/functions/is_blank.rb (hunk 1 above)
Comment (ozzyaaron) [inline lib/soroban/functions/is_blank.rb:1]: Nice spot. ISBLANK( “ “ ) should be false.

**Comment by kranzky** [comment]
:+1:

BTW, remember that this is a public open-source project, so all code and comments are visible to the world :)

**Comment by kranzky** [comment]
And wrt to that, we should support all ruby versions, so please don't break this: https://travis-ci.org/agworld/soroban/builds/22005793

**Comment by ozzyaaron** [comment]
I’m closing this as we will run a branch that coerces values until we can revisit a grammar based fix.

**Pull request closed without merging**